code/modules/mapping/mapping_helpers.dm

```diff
@@ -60,7 +60,7 @@
 	var/late = FALSE
 
 /obj/effect/mapping_helpers/Initialize(mapload)
-	..()
+	. = ..()
 	return late ? INITIALIZE_HINT_LATELOAD : qdel(src) // INITIALIZE_HINT_QDEL <-- Doesn't work
```

Comment:
Ditto.

Also can you check the claim in that comment?

Reply:
Checked.
This is a mess.
TL;DR: the claim is wrong.
TS;PE:
There are two direct children of `/obj/effect/mapping_helpers/` - `/airlock`, and `/no_lava`.
`airlock` has `var/late` set to `TRUE` so the parent returns `INITIALIZE_HINT_LATELOAD`. Then `airlock/Initialize()` promptly ignores it and returns its own `INITIALIZE_HINT_QDEL` unconditionally. It doesn't have LateInitialize either.

`no_lava` doesn't even use Initialize - it's a plain `New`.

The qdel works just fine on both of those.